AndroidSDKCore/src/main/java/com/leanplum/ActionContext.java

```diff
@@ -390,6 +390,7 @@ private boolean createActionContextForMessageId(String messageAction,
           public void variablesChanged() {
             try {
               ActionManager.getInstance().recordMessageImpression(actionContext.getMessageId());
+              Leanplum.triggerMessageDisplayed(actionContext);
```

Comment:
All three of these happen at the same place as all calls to `ActionManager.getInstance().recordMessageImpression`, so it seems they are inexorable tied together. I think it would make more sense to:
1. Put this inside `ActionManager.getInstance().recordMessageImpression`
2. Make a function that contains both the new callback and `ActionManager.getInstance().recordMessageImpression`.

I don't like the idea of keeping two functions in sync manually.

Reply:
Agreed.
I thought it made more sense to keep the `triggerMessageDisplayed` and move the `recordMessageImpression` under that. The `triggerMessageDisplayed` is now responsible for recording impression and calling any user callbacks.